rb_ws/src/buggy/scripts/auton/buggystate.py

```diff
@@ -0,0 +1,113 @@
+import utm
+from nav_msgs.msg import Odometry as ROSOdom
+from sensor_msgs.msg import NavSatFix
+
+from tf.transformations import euler_from_quaternion
+
+from std_msgs.msg import Bool
+
+from microstrain_inertial_msgs.msg import GNSSFixInfo
+
+
+
+import rospy
+
+
+class BuggyState:
+    """
+    Basically a translator from ROSOdom to ensure everything is in the correct units.
+
+    Other files should ONLY interface with the nav/odom messages through this file!
+
+    (functionally: this should replace telematics, pose and world)
+
+    """
+
+    def __init__(self, name = "sc"):
+        self.filter_odom : ROSOdom = None
+        self.gnss_1 : ROSOdom = None
+        self.gnss_2 : ROSOdom = None
+        self.gps_fix : int = 0
+
+
+        # to report if the filter position has separated (so we need to break)
+        rospy.Subscriber(name + "/debug/filter_gps_seperation_status", Bool, self.update_use_gps)

```

Comment:
Was this only for when we were going to switch between filter and GPS? Because if so, doesn't this mean it is dead code now?

Reply:
I feel like we should still know when the separation occurs, for status lights + debugging (so human driver knows buggy is messed up and needs to brake)

Reply:
Sounds good to me, so this is the buggy state, primarily used by debugging right? The comment looks fine, so this is probably good.